lib/internal/modules/esm/get_format.js

```diff
@@ -53,12 +64,28 @@ function getFileProtocolModuleFormat(url, context, ignoreErrors) {
   if (experimentalSpecifierResolution !== 'node') {
     // Explicit undefined return indicates load hook should rerun format check
     if (ignoreErrors) return undefined;
-    throw new ERR_UNKNOWN_FILE_EXTENSION(ext, fileURLToPath(url));
+    let suggestion = '';
+    if (getPackageType(url) === 'module' && ext === '') {
+      const config = getPackageScopeConfig(url);
+      const fileBasename = basename(filepath);
+      suggestion = 'Extension-less files are disabled inside of ' +
```

Comment:
Disabled or unsupported?

Reply:
We could change it to unsupported but that makes it seem like they will be supported sometime to me, last we talked about this it got stalled again.

Reply:
I was thinking "disabled" means it exists but is blocked by a switch that's off, whereas "unsupported" means it doesn't exist. My concern is people will hear the former and ask for the switch to be turned on.

Perhaps "disallowed"? 🤔 ("unsupported" still sounds better to me though)